lib/chef/knife/bootstrap.rb

```diff
@@ -696,9 +696,19 @@ def do_connect(conn_options)
         @connection = TrainConnector.new(host_descriptor, connection_protocol, conn_options)
         connection.connect!
       rescue Train::UserError => e
+        limit ||= 1
         if !conn_options.key?(:pty) && e.reason == :sudo_no_tty
           ui.warn("#{e.message} - trying with pty request")
           conn_options[:pty] = true # ensure we can talk to systems with requiretty set true in sshd config
+          retry
+        elsif config[:use_sudo_password] && (e.reason == :sudo_password_required || e.reason == :sudo_root_password_required) && limit < 3
+          ui.warn("Failed to authenticate #{conn_options[:user]} to #{server_name} - #{e.message} \n sudo: #{limit} incorrect password attempt")
+          sudo_password = ui.ask("Enter sudo root password for #{conn_options[:user]}@#{server_name}:") do |q|
```

Comment:
should probably just be `sudo password` not `sudo root password` to avoid confusion with the `root password` which is (hopefully) different.

Reply:
I have put the sudo root here due to 
in ubuntu while running `sudo su vagrant`
it raises

```
sudo su vagrant
[sudo] password for root: 
Sorry, try again.
[sudo] password for root:
```